cli/start.go

```diff
@@ -101,12 +102,36 @@ func initCacheSize() {
 	}
 }
 
+func initInsecure() error {
+	if !cliContext.Insecure || insecure.isSet {
+		return nil
+	}
+	// The --insecure flag was not specified on the command line, verify that the
+	// host refers to a loopback address.
+	if connHost != "" {
+		addr, err := net.ResolveIPAddr("ip", connHost)
+		if err != nil {
+			return err
+		}
+		if !addr.IP.IsLoopback() {
+			return fmt.Errorf("specify --insecure to listen on external address %s", connHost)
+		}
+	} else if connHost == "" {
```

Comment:
Because I wanted to increase my PR stats: https://github.com/cockroachdb/cockroach/pull/5631